File: front/src/api/types/project.ts
export interface IProject {
  _id: string
  id: string
  category: string
  title: string
  description: string
  slug: string
  cover: string
  tags: string[]
  createdAt: Date
  updatedAt: Date
  __v: number
}

export interface IProjectResponse {
  status: string
  data: {
    project: IProject
  }
}

export interface IProjectsResponse {
  status: string
  results: number
  total_results: number
  total_pages: number
  projects: IProject[]
}

File: back/src/models/project.rs
use chrono::prelude::*;
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Deserialize, sqlx::FromRow, Serialize, Clone)]
pub struct ProjectModel {
    pub id: Uuid,
    pub title: String,
    pub description: String,
    pub cover: String,
    pub slug: String,
    pub category: String,
    pub tags: Vec<String>,
    #[serde(rename = "createdAt")]
    pub created_at: Option<DateTime<Utc>>,
    #[serde(rename = "updatedAt")]
    pub updated_at: Option<DateTime<Utc>>,
}
